picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -745,13 +754,11 @@ public Builder withCacheSize(int maxByteCount) {
       if (maxByteCount < 0) {
         throw new IllegalArgumentException("maxByteCount < 0: " + maxByteCount);
       }
-      if (cache != null) {
-        throw new IllegalStateException("Memory cache already set.");
-      }
+      cacheSet = true;
       if (maxByteCount == 0) {
-        cache = Cache.NONE;
+        cache = null;
       } else {
-        cache = new LruCache(maxByteCount);
+        cache = new PlatformLruCache(maxByteCount);
```

Comment:
How about always creating a cache (even of size 0), instead of making the cache nullable and requiring null checks everywhere?  Adds less noise to this change.

Reply:
I considered it but was slightly concerned about overhead in ever insertion call for no cache.
That said, it's rare somebody doesn't want a cache.

Actually, could we cheat by checking maxSize and no-oping the cache `set` call?

Reply:
I don't think it's worth it unless we measure it to be a problem.

Reply:
yep, sounds good.
i was hoping you could do skip the cache with a RequestTransformer that changes the memory policy on all requests, but you can't.

anyway, will update tonight.

Reply:
OH SHOOT. android.util.LruCache doesn't let us have 0 for the size.

Reply:
went with making a 1-byte cache. we can always wait for somebody to complain if it breaks a use case.